client/src/actions/team.js

```diff
@@ -1,15 +1,70 @@
 /**
  * This file houses our team-related Action Creators.
  */
-import * as api from '../api';
+import * as api from "../api";
 import {
-    LINK_TEAM_TWITTER, 
-    UNLINK_TEAM_TWITTER, 
-    FETCH_TEAM_INFO,     
-    GET_TEAM_MEMBERS_BY_TEAM_ID, 
-    CREATE_TEAM_MEMBER, 
-    UPDATE_TEAM_MEMBER, 
-    DELETE_TEAM_MEMBER } from './types';
+  LINK_TEAM_TWITTER,
+  UNLINK_TEAM_TWITTER,
+  FETCH_TEAM_INFO,
+  GET_TEAM_MEMBERS_BY_TEAM_ID,
+  CREATE_TEAM_MEMBER,
+  UPDATE_TEAM_MEMBER,
+  DELETE_TEAM_MEMBER,
+  ADD_TEAM,
+} from "./types";
+
+/**
+ * Adds a new team to redux store and database. If api request fails the team state's requestFailed attribute will be 'true'
+ * @param teamInfo contains teamName, orgName and emailnot
+ */
+export const addTeamInfo = (teamInfo, result) => async (dispatch) => {
+  try {
+    const data = await api.addTeam(teamInfo);
+    const teamData = {
+      ...teamInfo,
+      teamId: data.data._id,
+      requestFailed: false,
+      error: "",
+    };
+    dispatch({
+      type: ADD_TEAM,
+      payload: teamData,
+    });
+    result(true, "");
+  } catch (err) {
+    result(false, err);
+  }
+};
+
+/**
+ * Finds a team
+ * @param teamCredentials team email and password as a dictionary
+ * @param teamPassword team account password
+ */
+export const getTeam = (teamCredentials, result) => async (dispatch) => {
+  try {
+    const data = await api.loginTeam(teamCredentials);
+    const teamData = data.data.team;
+    console.log(teamData);
+    const team = {
+      teamId: teamData._id,
+      email: teamData.email,
+      teamName: teamData.teamName,
+      orgName: teamData.orgName,
+      twitterHandle: teamData.twitterHandle,
+      repoCreated: teamData.repoCreated,
+      requestFailed: false,
```

Comment:
Should requestFailed and error be in the initial team state in teamReducer.js?

Reply:
These are removed now, not needed